doc/contributing.md

```diff
@@ -90,3 +90,109 @@ Sorry, no pseudonyms or anonymous contributions are allowed.
 
 We also require each commit be individually signed-off by their author,
 even when part of a larger set. You may find `git commit -s` useful.
+
+## Getting Started Developing
+
+Follow the steps below to set up your development environment to get started working on new features for LXD.
+
+### Dependencies
```

Comment:
For that section, it may be best to link to `index.md` rather than duplicate the information as we're bound to forget to update one of them.

The development specific parts about branches makes sense to keep in contributing though.
You may also want to mention `LXD_DIR` in the last section as an alternative to having your `/var/lib/lxd` modified by your fork of LXD.

Reply:
Thanks for the feedback @stgraber! I've incorporated your feedback and improved formatting of the "Important Notes" section.